nodedb.go

```diff
@@ -951,6 +804,55 @@ func (ndb *nodeDB) decrVersionReaders(version int64) {
 	}
 }
 
+// traverseOrphans traverses orphans which removed by the updates of the version (n+1).
+func (ndb *nodeDB) traverseOrphans(version int64, fn func(*Node) error) error {
+	cRoot, err := ndb.getRoot(version + 1)
+	if err != nil {
+		return err
+	}
+
+	curIter, err := NewNodeIterator(cRoot, ndb)
+	if err != nil {
+		return err
+	}
+
+	pRoot, err := ndb.getRoot(version)
+	if err != nil {
+		return err
+	}
+	prevIter, err := NewNodeIterator(pRoot, ndb)
+	if err != nil {
+		return err
+	}
+
+	var orgNode *Node
+	for prevIter.Valid() {
+		for orgNode == nil && curIter.Valid() {
+			node := curIter.GetNode()
+			if node.version <= version {
+				curIter.Next(true)
+				orgNode = node
+			} else {
+				curIter.Next(false)
+			}
+		}
+		pNode := prevIter.GetNode()
+
+		if orgNode != nil && bytes.Equal(pNode.hash, orgNode.hash) {
```

Comment:
is it more efficient to do an extra pointer comparison here: `pNode == orgNode`

Reply:
that's a good point, if they are from cache